lang/java/reef-webserver/src/main/java/org/apache/reef/webserver/HttpServerReefEventHandler.java

```diff
@@ -169,20 +171,22 @@ public void onHttpRequest(
       writeLines(response, result, "Current Stages...");
       break;
     case "logfile":
-      final List names = parsedHttpRequest.getQueryMap().get("filename");
+      final List<String> names = parsedHttpRequest.getQueryMap().get("filename");
+      final PrintWriter writer = response.getWriter();
       if (names == null || names.size() == 0) {
-        response.getWriter().println(String.format("File name is not provided"));
-      }
-
-      final String fileName = (String)names.get(0);
-      if (!fileName.equals(driverStdoutFile) && !fileName.equals(driverStderrFile)) {
-        response.getWriter().println(String.format("Unsupported file names: [%s] ", fileName));
-      }
-      try {
-        final byte[] outputBody = readFile((String) names.get(0)).getBytes(StandardCharsets.UTF_8);
-        response.getOutputStream().write(outputBody);
-      } catch(final IOException e) {
-        response.getWriter().println(String.format("Cannot find the log file: [%s].", fileName));
+        writer.println("File name is not provided");
+      } else {
+        final String fileName = names.get(0);
+        if (!fileName.equals(driverStdoutFile) && !fileName.equals(driverStderrFile)) {
+          writer.println(String.format("Unsupported file names: [%s] ", fileName));
+        } else {
+          try {
+            final byte[] outputBody = readFile(fileName).getBytes(StandardCharsets.UTF_8);
+            writer.print(Arrays.toString(outputBody));
+          } catch (final IOException e) {
+            writer.println(String.format("Cannot find the log file: [%s].", fileName));
```

Comment:
There can be 2 causes of `IOException` here - the first one is failing to write to the `response` stream (if you add back the line) and the second is failing to find the log file. The error message may need to account for both.